app/leaderboard/Leaderboard.tsx

```diff
@@ -3,21 +3,44 @@
 import LeaderboardCard from "@/components/contributors/LeaderboardCard";
 import { TbZoomQuestion } from "react-icons/tb";
 import TopContributor from "../../components/contributors/TopContributor";
-import { useState } from "react";
+import { useEffect, useState } from "react";
 import { usePathname, useRouter, useSearchParams } from "next/navigation";
 import { getWeekNumber, parseDateRangeSearchParam } from "@/lib/utils";
 import DateRangePicker, { formatDate } from "@/components/DateRangePicker";
 import Search from "@/components/filters/Search";
 import Sort from "@/components/filters/Sort";
+import RoleFilter from "@/components/filters/RoleFilter";
 import format from "date-fns/format";
-import { LeaderboardAPIResponse } from "../api/leaderboard/functions";
+import { LeaderboardAPIResponse } from "@/app/api/leaderboard/functions";
+import { SelectOption } from "@/components/Select";
 
 export default function Leaderboard(props: { data: LeaderboardAPIResponse }) {
   const router = useRouter();
   const pathname = usePathname();
   const searchParams = useSearchParams();
   const [searchTerm, setSearchTerm] = useState("");
   const [start, end] = parseDateRangeSearchParam(searchParams.get("between"));
+  const [roleFilter, setRoleFilter] = useState<SelectOption[]>(
+    searchParams
+      .get("role")
+      ?.split(",")
+      .map((value) => ({
+        value,
+        text: FILTER_BY_ROLE_OPTIONS[
+          value as keyof typeof FILTER_BY_ROLE_OPTIONS
+        ],
+      })) || [],
+  );
+  const [sortBy, setSortBy] = useState<SelectOption | undefined>(
+    searchParams?.get("sortBy")
+      ? ({
+          value: searchParams.get("sortBy"),
+          text: SORT_BY_OPTIONS[
+            searchParams.get("sortBy") as keyof typeof SORT_BY_OPTIONS
+          ],
+        } as (typeof SortOptions)[number])
+      : { value: "points", text: "Points" },
+  );
```

Comment:
Can we not use "useState" and "useEffect" here? 

When we update the filter selections, let it directly update the search params. This will cause the values returned by useSearchParams to be updated anyways. Essentially moving the state management to the url params.